### PR TITLE
[#1] Setup project

### LINKED ISSUE #1: Setup project
## Why

Setup base application from scratch.

## Acceptance Criteria

- Setup [Next.js](https://nextjs.org/)

### PR DESCRIPTION
**Pull request by Lahphim**
* Resolve [Setup project #1](https://github.com/Lahphim/react-surveyor-challenge/issues/1)

## What happened 👀

Setup base application from scratch.

## Insight 📝

- Setup [Next.js](https://nextjs.org/)
- Support TypeScript

## Proof Of Work 📹

| Run the application |
| ------------------- |
| Screen Shot 2565-05-10 at 18 13 24 |

| Web application |
| ---------------- |
| Screen Shot 2565-05-10 at 18 13 52 |

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "next/core-web-vitals"
```
Comment (hoangmirs) [inline .eslintrc.json]: Do you have a plan to add our ESlint config? 😄 
You can put our config at the top:
```suggestion
  "extends": [
    "@nimblehq/eslint-config-nimble-react",
    "next/core-web-vitals"
  ]
```

File: pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import type { NextPage } from 'next'
+import Head from 'next/head'
+import Image from 'next/image'
+import styles from '../styles/Home.module.css'
+
+const Home: NextPage = () => {
```
Comment (hoangmirs) [inline pages/index.tsx]: Should we move `pages` and `styles` folder into `src` folder to make it's similar to other projects?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "strict": true,
+    "forceConsistentCasingInFileNames": true,
+    "noEmit": true,
+    "esModuleInterop": true,
+    "module": "esnext",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "jsx": "preserve",
+    "incremental": true
```
Comment (hoangmirs) [inline tsconfig.json]: If you agree on creating `src` folder, you can add this line here:
```
"baseUrl": "./src"
```

**Review by hoangmirs — APPROVED** [review]
LGTM as this is generated from the `create-next-app` CLI. I just left a few comments

File: README.md [hunk 4]
```diff
@@ -1,12 +1,41 @@
...
+* Rating
+* NPS
+* Text input
+* Textarea input
+* Select option
+* Multiple choice
+* Slider
+
+## Prerequisite
+* [Next.js](https://github.com/vercel/next.js)
```
Comment (carryall) [inline README.md:19]: Maybe add the node version here too? 💭

**Review by carryall — APPROVED** [review]

File: pages/index.tsx (hunk 2 above)
Comment (Lahphim) [reply pages/index.tsx]: Moved those folders into `src` https://github.com/Lahphim/react-surveyor-challenge/pull/44/commits/f7bed671df20dc7da86f66821803a5d2dcbd0084

File: tsconfig.json (hunk 3 above)
Comment (Lahphim) [reply tsconfig.json]: Added new `baseUrl` https://github.com/Lahphim/react-surveyor-challenge/pull/44/commits/f7bed671df20dc7da86f66821803a5d2dcbd0084

File: README.md (hunk 4 above)
Comment (Lahphim) [reply README.md:19]: Added node into the list of prerequisites https://github.com/Lahphim/react-surveyor-challenge/pull/44/commits/7304117282ca348b8a3dcc5bcadd5629fb634ada

File: .eslintrc.json (hunk 1 above)
Comment (Lahphim) [reply .eslintrc.json]: Used our linter https://github.com/Lahphim/react-surveyor-challenge/pull/44/commits/fb342bdc39983cb2699c0cbced853cf8b9f73227

**Pull request merged**